lib/serverrequestcontext.go

```diff
@@ -188,11 +188,14 @@ func (ctx *serverRequestContextImpl) verifyX509Token(ca *CA, authHdr, method, ur
 	if err2 != nil {
 		return "", caerrors.NewAuthenticationErr(caerrors.ErrInvalidToken, "Invalid token in authorization header: %s", err2)
 	}
+
 	// Make sure the caller's cert was issued by this CA
-	err2 = ca.VerifyCertificate(cert)
+	reenrollIgnoreCertExpiry := ctx.endpoint.Path == "reenroll" && ctx.ca.Config.CA.ReenrollIgnoreCertExpiry
```

Comment:
The existing comment right above this line doesn't go with this line so it is a bit confusing.
I'd keep the existing comment with the ca.VerifyCertificate() line, and add a separate comment for this new line explaining that if it is true then we'll forceTime in the VerifyCertificate() check.